src/ViewItem.js

```diff
@@ -99,10 +100,11 @@ class ViewItem extends React.Component {
       params: (_q, _p, _r, _l, props) => {
         const servicePointId = get(props.resources, 'items.records[0].lastCheckIn.servicePointId', '');
 
-        if (!servicePointId) return {};
+        const query = servicePointId && `id==${servicePointId}`;
 
-        return { query: `id==${servicePointId}` };
+        return query ? { query } : null;
```

Comment:
So, this is totally fine in normal use and it's actually the way I wrote my code at first. 

I was later writing some BigTests and ran into issues in my Mirage code where it didn't nicely handle a `null` search string. This is why I've been transitioning my code to a `return query ? { query } : {}` pattern instead.

There may be a better way of writing Mirage handlers that obviates the need for returning an empty object here, in which case I'd recommend using your current, `null`-returning pattern instead. Or if you're not planning on using BigTest w/ Mirage handlers for this endpoint, then feel free to ignore this as well.

Reply:
Thanks for your advice about using BigTest with Mirage handlers. I will take this case in mind.
The thing is using an empty-object-returning pattern here, when the entity ID is missing, leads to the receiving of all entities. While using a null-returning pattern does not send a request at all, which meets the requirements of this story.